b2sdk/b2http.py

```diff
@@ -235,34 +244,77 @@ def post_content_return_json(
            except B2Error as e:
                ...
 
-        :param str url: a URL to call
-        :param dict headers: headers to send.
-        :param data: bytes (Python 3) or str (Python 2), or a file-like object, to send
-        :return: a dict that is the decoded JSON
-        :rtype: dict
+        :param method: HTTP method: 'POST' or 'GET'
+        :param url: a URL to call
+        :param headers: headers to send.
+        :param data: raw bytes or a file-like object to send
+        :param try_count: a number of retries
+        :param params: a dict that will be converted to query string for GET requests or additional metadata for POST requests
+        :param stream: if True, the response will be streamed
+        :return: final response
+        :raises: B2Error if the request fails
         """
+        method = method.upper()
         request_headers = {**headers, 'User-Agent': self.user_agent}
 
-        # Do the HTTP POST.  This may retry, so each post needs to
-        # rewind the data back to the beginning.
-        def do_post():
-            data.seek(0)
-            self._run_pre_request_hooks('POST', url, request_headers)
-            response = self.session.post(
+        def do_request():
+            # This may retry, so each time we need to rewind the data back to the beginning.
+            if data is not None and not isinstance(data, bytes):
```

Comment:
Use [seekable()](https://docs.python.org/3/library/io.html#io.IOBase.seekable)?

Reply:
I would prefer not to allow more types than it previously did allow (in practice, docstring was all wrong).
I added bytes handling since it seemed to make some kind of sense at least to have.

When someone finds a use for detecting if file is seekable or not let have them add it.

As to why I don't want to go into that rabbit hole:
If you have FIFO input that is not seekable then you basically CANNOT do retries - I don't think we want to have something like that handled here.